From d6877a7c1c35985f6a75b6cd4e814595e781adc4 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Wed, 21 Oct 2015 21:03:00 -0700
Subject: [PATCH 05/10] Use NO_WEAK_SYMBOLS instead of -flat_namespace

Lesser of two evil hacks, I suppose...

This reverts commit 0386fa77367a305deea3cc27f8a3865cc3c467c0.
---
 configure.ac      |  3 ---
 src/stubs/stubs.h | 11 +----------
 2 files changed, 1 insertion(+), 13 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7cb6bdf..640cc92 100644
--- a/configure.ac
+++ b/configure.ac
@@ -213,9 +213,6 @@ case $host_os in
     linux*)
         OS_CFLAGS="-D_XOPEN_SOURCE"
         ;;
-    darwin*)
-        OS_CFLAGS="-Wl,-flat_namespace"
-	;;
     *)
         OS_CFLAGS=
         ;;
diff --git a/src/stubs/stubs.h b/src/stubs/stubs.h
index c09c675..9e71806 100644
--- a/src/stubs/stubs.h
+++ b/src/stubs/stubs.h
@@ -10,7 +10,7 @@
 #endif
 
 /* this probably works for Mach-O too, but probably not for PE */
-#if (defined(__APPLE__) || defined(__ELF__)) && defined(__GNUC__) && (__GNUC__ >= 3)
+#if defined(__ELF__) && defined(__GNUC__) && (__GNUC__ >= 3)
 #define weak __attribute__((weak))
 #else
 #define weak
@@ -67,15 +67,6 @@ extern void (*__register_fpe_functions)(void);
 #define OVERRIDE_VA_SYMBOL(sym,f)
 #endif
 
-/* This is really just a hack for now... __APPLE__ really should be using
- * the weak symbols route above, but it's causing an as-yet unresolved issue,
- * so we're instead building with flat_namespace.
- */
-#ifdef __APPLE__
-#undef weak
-#define weak
-#endif
-
 extern FontPtr find_old_font ( FSID id );
 extern int set_font_authorizations ( char **authorizations,
 				     int *authlen,
-- 
2.8.3

